agent/structs/config_entry.go

```diff
@@ -159,11 +162,83 @@ func (e *ProxyConfigEntry) GetRaftIndex() *RaftIndex {
 	return &e.RaftIndex
 }
 
+func (c *ProxyConfigEntry) MarshalBinary() (data []byte, err error) {
+	type alias ProxyConfigEntry
+
+	a := alias(*c)
+
+	// bs will grow if needed but allocate enough to avoid reallocation in common
+	// case.
+	bs := make([]byte, 128)
+	enc := codec.NewEncoderBytes(&bs, msgpackHandle)
+	err = enc.Encode(a)
+	if err != nil {
+		return nil, err
+	}
+
+	return bs, nil
+}
+
+func (e *ProxyConfigEntry) UnmarshalBinary(data []byte) error {
```

Comment:
super tiny nitpick, the pointer receiver is named `e` here but `c` above in `MarshalBinary`